### PR TITLE
Fix issues related to empty stack dir, optional cluster_name_pattern and better of use_eks flag

### PR DESCRIPTION
**Pull request by shahincsejnu**
## what

1. atmos helmfile commands should not require EKS
2. cluster_name_pattern should only use for EKS
3. website url link fixing

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling for AWS configurations when using EKS.
	- Improved validation logic for stack manifests, ensuring necessary files are present before execution.
	
- **Bug Fixes**
	- Corrected error message formatting for improved clarity.
	- Updated documentation for vendoring to clarify functionality and benefits.

- **Documentation**
	- Expanded use cases and features related to vendoring in Atmos.
	- Updated links in documentation for better navigation. 

- **Chores**
	- Added new entries to `.gitignore` to streamline version control. 
	- Updated various dependencies to improve compatibility and performance.

### CODE REVIEW

File: internal/exec/helmfile.go [hunk 1]
```diff
@@ -275,11 +278,6 @@ func checkHelmfileConfig(cliConfig schema.CliConfiguration) error {
 			return errors.New("Helm AWS profile pattern must be provided in 'components.helmfile.helm_aws_profile_pattern' config or " +
 				"'ATMOS_COMPONENTS_HELMFILE_HELM_AWS_PROFILE_PATTERN' ENV variable")
 		}
-
-		if len(cliConfig.Components.Helmfile.ClusterNamePattern) < 1 {
-			return errors.New("Cluster name pattern must be provided in 'components.helmfile.cluster_name_pattern' config or " +
```
Comment (osterman) [inline internal/exec/helmfile.go:280]: @aknysh do we still require this if use_eks is true?

**Comment by aknysh** [comment]
@shahincsejnu  thanks, please address the comments and resolve the conflicts

File: internal/exec/helmfile.go (hunk 1 above)
Comment (osterman) [reply internal/exec/helmfile.go:280]: @shahincsejnu we should require `cluster_name_pattern` if `use_eks` is true

Comment (shahincsejnu) [reply internal/exec/helmfile.go:280]: @osterman okay, made it required!

**Comment by shahincsejnu** [comment]
> @shahincsejnu thanks, please address the comments and resolve the conflicts

@aknysh resolved & updated, pls have a look again, thanks!

**Comment by osterman** [comment]
There are new merge conflicts. Also, please show screenshots validating the functionality against k3s.

**Pull request closed without merging**